src/itree.cpp

```diff
@@ -0,0 +1,133 @@
+#include "rodsClient.h"
+#include "client_connection.hpp"
+#include <boost/program_options.hpp>
+#include "irods_client_api_table.hpp"
+#include "irods_pack_table.hpp"
+#include "irods_parse_command_line_options.hpp"
+#include "irods/filesystem.hpp"
+
+#include "fmt/core.h"
+#include "fmt/color.h"
+
+namespace po = boost::program_options;
+namespace fs = irods::experimental::filesystem;
+auto print_dir(const fs::path&,
+               const po::variables_map&,
+               irods::experimental::client_connection&) -> void;
+
+auto correct_path(const po::variables_map&,const rodsEnv&)->fs::path;
+
+// ANSI escape numbers. For setting the colors.
+static const auto directory_color=fmt::color::blue;
+static const auto file_color = fmt::color::green;
+
+int main(int argc, char** argv){
+    rodsEnv env;
+    int status = getRodsEnv(&env);
+    irods::experimental::client_connection conn;
+    po::options_description desc{""};
+    po::positional_options_description pod;
+    pod.add("dir",1);
+    desc.add_options()
+        ("help","Display help")
+        ("dir",po::value<std::string>()->default_value(env.rodsCwd), "The collection to start from(defaults to the current working collection)")
+        ("directories_only,d", po::bool_switch(),"Only list directories")
+        ("fullpath,f",po::bool_switch(),"print the full path of each item listed")
+        ("depth,L", po::value<int>()->default_value(1000), "Limit the depth of the listing.")
+        ("color,c", po::bool_switch(), "Print in color(requires ansi terminal)");
+    if ( status < 0) {
+        std::cerr << "Irods connection error" << std::endl;
```

Comment:
I'm not sure this is a connection issue at this point.

Reply:
I think @korydraughn has had opinions about `std::endl` usage before.    too much flushing?

Reply:
That is correct. You don't want to use it in code that is called frequently. It will degrade performance if used in hot paths.